src/state/charm_state.py

```diff
@@ -125,7 +128,9 @@ def _validate_state(
         is_ingress_related = bool(ingress_provider.relations)
         is_ingress_per_unit_related = bool(ingress_per_unit_provider.relations)
         is_legacy_related = bool(reverseproxy_requirer.relations)
-        is_haproxy_route_related = bool(haproxy_route_provider.relations)
+        is_haproxy_route_related = bool(
```

Comment:
I don't see any unit tests changes for this change

Reply:
added